File: ContainsDuplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) 
    {
        unordered_map<int, int> map;
        
        int size = nums.size();
        
        for(int i = 0; i < size; i++)
        {
            if(map.find(nums[i]) != map.end())
            {
                return true;
            }
            
            map[nums[i]]++;
        }
        
        return false;
    }
};

File: ContainsDuplicate.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        
        a = {}
        
        for i in range(len(nums)):
            
            if nums[i] in a:
                
                if(i - a[nums[i]] <= k):
                    
                    return True
                
            a[nums[i]] = i
        
        return False
            
            